builtins/amp-video.js

```diff
@@ -36,7 +36,31 @@ export function installVideo(win) {
     }
 
     /** @override */
+    prerenderAllowed() {
+      return true;
+    }
+
+    /** @override */
+    buildCallback() {
+      /** @private */
+      this.realPreloadAttrValue_ = this.element.getAttribute('preload');
+      const posterAttr = this.element.getAttribute('poster');
+
+      if (!posterAttr) {
+        console/*OK*/.warn(
+            'No "poster" attribute has been provided for amp-video.');
+      }
+    }
+
+    /** @override */
     layoutCallback() {
+      const docState = documentStateFor(this.getWin());
+      // Disable video preload in prerender mode and reset it when rendered.
+      if (docState.getVisibilityState() === 'prerender') {
+        this.element.setAttribute('preload', 'none');
```

Comment:
`layoutCallback` is not called again, unless the element opts-in. I'd say it'd be somewhat dangerous, but definitely possible. Instead, I'd rather introduce a separate event for these cases.